Legacy/bonej/src/main/java/org/bonej/plugins/ParticleAnalysis.java

```diff
@@ -718,20 +718,25 @@ private static double getSurfaceArea(final List<Point3f> points) {
 	 * algorithm.
 	 *
 	 * @param surfacePoints points of all the particles.
-	 * @return Feret diameters of the surfaces.
+	 * @return Feret diameters and x, y, z coordinates of the two feret points of each surface,
+	 * packed in a double so that the feret diameter of particle i is found at [i * 7] and the
+	 * points' coordinates are in the following 6 positions in ax, ay, az, bx, by, bz order.
 	 */
 	static double[] getFerets(final List<List<Point3f>> surfacePoints) {
 		Thread[] threads = Multithreader.newThreads();
 		final int nSurfaces = surfacePoints.size();
-		final double[] ferets = new double[nSurfaces];
+		//distance, xa, ya, za, xb, yb, zb
+		final double[] ferets = new double[nSurfaces * 7];
```

Comment:
Would be clearer with a `final double[][] ferets = new double[nSurfaces][7];`. This does break API, but that's not a problem since the method is package-private (and only used in one place).

Reply:
Agreed, thanks.